### PR TITLE
Include md files from MarkdownFiles Directory field when publishing

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Per https://jira.autodesk.com/browse/DYN-5447, when publishing the current package, include Markdown files from MarkdownFiles Directory field.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Include md files from MarkdownFiles Directory field when publishing

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs [hunk 1]
```diff
@@ -655,6 +655,11 @@ public ObservableCollection<string> AdditionalFiles
             }
         }
 
+        /// <summary>
+        /// Optional Markdown files list
+        /// </summary>
+        internal List<string> MarkdownFiles;
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs:661]: seems this should be initialized to an empty list - though why is this a list and not an array, Ienumerable etc - I don't see it being used as a list (dynamic adding removing etc)

**Comment by mjkkirschner** [comment]
one real failing test:
https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/7808/testReport/junit/DynamoCoreWpfTests/PublishPackageViewModelTests/CanPublishLateInitializedJsonCustomNode/

File: src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs (hunk 1 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs:661]: Good point, Ienumerable or array should both work, will update

**Review by mjkkirschner — APPROVED** [review]

**Comment by mjkkirschner** [comment]
If you have time it would be good to update the package builder tests to add tests for the docs folder.

**Comment by QilongTang** [comment]
> If you have time it would be good to update the package builder tests to add tests for the docs folder.

Sure. Done

**Pull request merged**